src/tls.c

```diff
@@ -7160,7 +7164,7 @@ int TLSX_PopulateExtensions(WOLFSSL* ssl, byte isServer)
                 /* Pre-shared key is mandatory extension for resumption. */
                 ret = TLSX_PreSharedKey_Use(ssl, sess->ticket, sess->ticketLen,
                                             milli, ssl->specs.mac_algorithm, 1,
-                                            ssl->heap);
+                                            NULL);
```

Comment:
Embarrassingly the last parameter of the function is not for the heap but for the return of the pre-shared key object. The object is not needed and so the parameter is NULL.